composer/trainer/devices/device_gpu.py

```diff
@@ -29,6 +29,8 @@ class DeviceGPU(Device):
     dist_backend = 'nccl'
 
     def __init__(self, device_id: Optional[int] = None):
+        if not torch.cuda.is_available():
```

Comment:
nit: Do we need this check? I believe it will error below on line 36 if cuda is not available.

Reply:
Yes, but the error is a bit obscure, so a check here would be more informative.